### ISSUE TITLE
Modified files are not shown as such after switching focus

### ISSUE THREAD

**Issue by darQrrR** [atom/tree-view#906]
Created a new Issue as requested, the problem has already been there in previous atom-versons #725.

OS: Win10
Atom Versions: 1.9.0 and 1.9.1 

Reproduction:
1. Modify files in Atom
2. Files are shown correctly as modified in Tree-View
3. Switch Focus to any other Application
4. Switch back to Atom
5. Files are now all shown in grey

**Comment by arcticicestudio** [comment]
Can confirm this for version `1.9.2`.
OS: Arch Linux `4.6.4-1-ARCH`
Node: `6.3.1`
npm: `3.10.5`

**Comment by 50Wliu** [comment]
Thanks for creating a new issue @darQrrR!

Here are's @csarnataro's repro steps:
The same is happening to me on OS X Yosemite with Atom 1.9.2.

Versions:

OS: OS X Yosemite
Atom: 1.9.2
Reproduction steps:

prerequisites: having a project in the tree view under git control
open Atom. At this point the files in tree view are properly coloured regarding their git status (green for new files, orange/yellow for modified files and directory)
cmd+tab to switch another application
cmd+tab to come back to Atom
all files are grey
also, at this point, if a single file is modified, that only file changes its colour in the tree view
cmd+tab to switch another application
cmd+tab to come back to Atom
the modified file becomes grey

**Comment by 50Wliu** [comment]
For the record I cannot reproduce this using Atom 1.11.0-dev-c381b93.

**Comment by dbalcomb** [comment]
Just updated to 1.9.2 this morning from 1.8.x and it appears that I am having the same issue but not for all projects. I wouldn't describe the files in the tree view going grey (I suppose that would depend on the theme) but they do lose their git status information.

Additionally the git-status area in the status bar also disappears when the window regains focus so I am sceptical that this is an issue with the tree-view module.

I mentioned above that the issue is not happening across all of my git projects. For me it seems to specifically affect those projects that have a second non-git project folder. I have not yet tried adding additional project folders or adding a second git project folder however removing the second project folder seems to prevent the issue from occurring. I have also not attempted to reproduce the issue on a dev version of atom.

OS: OS X El Capitan
Atom: 1.9.2
Reproduction steps:
1. Open a git repository within atom.
2. Use File -> Add Project Folder... to add a second non-git project folder.
3. Make a change to a file within the repository and leave the file open within the editor.
4. Switch focus to another application.
5. Switch focus back to the atom window.

The tree view should have lost information about the git status and the status-bar should no longer have the section indicating the number of lines that have changed.

**Comment by 50Wliu** [comment]
> Make a change to a file within the repository

Which repository?  The Git one or the non-Git one?

**Comment by dbalcomb** [comment]
> Which repository? The Git one or the non-Git one?

Perhaps I should have been more clear. The git repository as it was the only one I referenced as a repository instead of a project folder. Changing the file is not what causes the problem as you could also add a new file. It was merely a step that would ensure the git repository would have a change that is not yet committed and so would show up in the tree view for demonstration purposes. Otherwise a git repository with no changes and no ignored files would not look any different.

**Comment by 50Wliu** [comment]
Confirmed.  Thanks for the additional reproduction steps @dbalcomb.

**Comment by orochies** [comment]
I have the same issue:

windows: 8.0
Atom: 1.9.2

Only happens when I add more of a git project in tree view. If I have only one project everything works fine.

**Comment by haleksandre** [comment]
Same focus issues while having multiple project folders in the tree-view, otherwise seems to be working fine.

OS: Ubuntu 16.04
Atom: 1.9.4

**Comment by code-tree** [comment]
I can confirm the problem still exists on version 1.10.0-beta2. I don't have a problem when I change window focus, but the problem does occur when multiple (git) projects are open in same window.

**Comment by nathansobo** [comment]
Able to reproduce. Investigating...

**Comment by nathansobo** [comment]
Found the problem. Waiting on builds but will try to get a fix out today. Sorry for the bumps and bruises on this release.

**Comment by nathansobo** [comment]
This will be fixed in 1.9.6. Thanks for your reports everyone.

**Comment by nathansobo** [comment]
1.9.6 is out with this fixed.

**Comment by marszall87** [comment]
Wow, that was fast 👍

**Comment by comzyh** [comment]
Finally. It fixed.

**Comment by schoettler** [comment]
@nathansobo Thanks guys! Although, now I'm getting a folder highlighted when a .gitignored file has changed

**Comment by int64ago** [comment]
What about this [issue](https://github.com/atom/atom/issues/4982), they are similar.

**Comment by 50Wliu** [comment]
@int64ago No, they are not.  This was a regression that dealt with multiple Git repositories being open.  Your referenced issue is about files not being refreshed upon switching Git branches.

**Comment by ryanmccarthypdx** [comment]
I am still having this problem on 1.9.9 on El Capitan 10.11.6

Repro steps are same as above:

1. Modify files in Atom
2. Files are shown correctly as modified in Tree-View
3. Switch Focus to any other Application
4. Switch back to Atom
5. Files are now all shown in grey

**Comment by baziliolg** [comment]
Atom 1.13.0 x64 on Debian Jessie.

If I have my whole home directory added as first item in tree view (just for convenience of navigating through notes, logs, etc.), all subsequent adds of project folders (which also reside in my home directory, but ) with Git lose colour markings of modified or added files.
As soon as I remove home directory from tree view or just move it by dragging __after__ other project folders, colour markings come back. This also affects colour marks of git-diff package.

**Comment by a7madgamal** [comment]
I have a question for another (but similar) issue
the tree view git status colors don't update if the window is unfocused
most common case: 
- I have changes in the tree
- I switch to sourcetree or the terminal to commit them
- the colors stay there untill I focus the window again
please let me know if this related to this issue or if should open a new one

**Comment by erkanarslan** [comment]
This problem still exists on Atom 1.16.0 on Ubuntu 16.04

**Comment by bitdawg** [comment]
I seem to have the opposite problem... if I'm in atom and a process runs that modifies files, the files don't show as modified (or added) until after I remove focus from atom and refocus.

Is there a way to refresh the tree view status without clicking away from atom?

**Comment by serpent213** [comment]
Similar problem here with Atom 1.14.4 on MacOS Sierra. Cannot find a connection to focus/unfocus. It did work for many days, then the problem appeared. The parent folder (not a git repo) was included as a project directory. I removed that from the tree and the colouring worked again.

**Comment by sebastianoc** [comment]
> Similar problem here with Atom 1.14.4 on MacOS Sierra. Cannot find a connection to focus/unfocus. It did work for many days, then the problem appeared. The parent folder (not a git repo) was included as a project directory. I removed that from the tree and the colouring worked again.

I confirm this workaround made the issue gone
